### ISSUE TITLE
The Blackboard

### ISSUE THREAD

**Issue by wchow** [kfcteh/genomereshuffled#2]
## Ideas & Random Musings

**Use this issue tracker to add new ideas, or wishlist**

**Comment by wchow** [comment]
Perhaps a potential workflow:
1. feed files in (gff3, agp)
2. reshuffle
3. return new agp file (ie new map of reshuffled components)

Not sure if this can all be done on the client side,  I'm sure its possible since its really all text.

**Comment by wchow** [comment]
Possibly first steps is to write a JS parser?

-Parse the files to create the objects for drawing.  
-the canvas will be the assembled object (ie Chr)
-then created individual component/parts object to map to this.
-then attach features of the individual components

what do you think?

**Comment by kfcteh** [comment]
I just modified the BCRail.agp text file to remove the gaps.  Do you think this file looks okay to parse?  I've managed to parse this file already using javascript on the client side.  
[BCRail.txt](https://github.com/kfcteh/genomereshuffled/files/531666/BCRail.txt)

**Comment by wchow** [comment]
Yeah man, that works, I think perhaps its best to built the framework first as proof of principal then maybe add the little trinkets such as gaps.  let me know what I can do to help.

**Comment by kfcteh** [comment]
Okay sounds good.  I think I'm okay for now.  Any pushes to master will automatically appear live in the Heroku website.

Currently, users are able to drag and drop the text file into a DIV in the browser window like gmail, drop box and the browser does the parsing client-side.  I'm going to use the pixie.js rendering engine to render the images client side using webGL. Once I finish these changes I'll push to master.  The web pages will be served to the client using Node.js Express web server.

https://genomereshuffled.herokuapp.com/

Maybe I will need some help later when we do the CSS HTML styling?

Let me know if you have any suggestions or ideas

Thanks

**Comment by wchow** [comment]
are u using node for heroku?

**Comment by kfcteh** [comment]
Yes I'm using node.  Play with it if you want.  Your pushes to master will show up live.

**Comment by wchow** [comment]
Hey man, 

I'm not familiar with some of the javascript library/modules.  Besides the node ones, what website are you using for the documentation on what to use/are using?

**Comment by wchow** [comment]
wait you're just using react right? so the docs should be there?

**Comment by kfcteh** [comment]
I'm using express for the web server.  
https://expressjs.com/

I haven't put in the React component yet.  I was just thinking for now to have a web server serve up the pages.  I was thinking to use javascript for the front and back end.

**Comment by kfcteh** [comment]
Sample data set for testing

[BCRail.txt](https://github.com/kfcteh/genomereshuffled/files/552533/BCRail.txt)

**Comment by wchow** [comment]
example of gamification of science dataset.
http://scienceblog.cancerresearchuk.org/2014/02/04/download-our-revolutionary-mobile-game-to-help-speed-up-cancer-research/
